app/src/main/java/com/chouten/app/data/WebviewHandler.kt

```diff
@@ -2,358 +2,291 @@ package com.chouten.app.data
-                    .replace('"', '\'')
-
-            // We want to buffer the sanitised data
-            // and print it to the logcat
-            // so that we can debug it.
-            val chunkSize = 1000
-            val chunks = sanitised.chunked(chunkSize)
-            chunks.forEachIndexed { index, chunk ->
-                Log.d("WebviewHandler", chunk)
-            }
+        if(req.action == "HTTPRequest" && req.url != null && req.headers != null){
+            
+            var responseText = "";
 
-            Log.d("WebviewHandler", "Injecting: $sanitised")
-            val skeleton = """
-                <html>
-                    <head>
-                    ${
-                codeblock.imports?.joinToString("") {
-                    val url = URLDecoder.decode(it, "UTF-8")
-                    "<script src=\"$url\"></script>"
-                } ?: ""
+            if(req.method == "POST"){
+                responseText = client.post(url=req.url, headers=req.headers, requestBody=req.body?.toRequestBody()).body.string();
+            }else{
+                responseText = client.get(req.url, req.headers).body.string();
             }
-                    </head>
-                    <body>
-                        <div id="chouten"></div>
-                        <div id="json-result" data-json="$sanitised"></div>
-                    </body>
-                </html>
-            """.trimIndent()
-            val skeletonEncoded = Base64.encodeToString(
-                skeleton.toByteArray(StandardCharsets.UTF_8),
-                Base64.DEFAULT
-            ).toString()
-            webview.loadData(skeletonEncoded, "text/html", "base64")
-        }
 
-        injectionLock.lock()
-
-        // We want the DOM to be loaded before we inject
-        // the data into the webview.
-        webview.webViewClient = object : WebViewClient() {
-            override fun onPageFinished(view: WebView?, url: String?) {
-                // Inject the data into the webview
-                // and then remove the webview client.
-                // Add the chouten div to the DOM
-                if (codeblock.usesApi) {
-                    injectionLock.unlock()
-                    return
-                }
-
-                webview.evaluateJavascript(
-                    """
-                    var chouten = document.createElement("div");
-                    chouten.id = "chouten";
-                    document.body.prepend(chouten);
-                    """.trimIndent()
-                ) {
-                    injectionLock.unlock()
-                }
+            val myWebView = this.webview;
+            val response: Map<String, String> = mapOf(
+                "reqId" to req.reqId,
+                "responseText" to responseText
+            );
+            
+            withContext(Dispatchers.Main) {
+                myWebView.postWebMessage(WebMessage(JSONObject(response).toString()), Uri.parse("*"))
```

Comment:
Same as previous comments referencing the use `Mapper.json.encodeToString`

Reply:
TODO added.